### DISCUSSION TITLE
I'm developing the user login part, with a database. However, I'm having difficulty with the login route, where it can't verify my password, what could I be doing wrong?

### DISCUSSION THREAD

**Discussion by JoaoLuiz92** [adonisjs/core#4548] [Help]
`router.post('login', async ({ request, response }) => {
  const { email, password } = request.only(['email', 'password'])

  const user = await User.findBy('email', email)
  if (!user) {
    return response.abort('Invalid')
  }

  const passwordValid = await hash.verify(user.password, password)
  console.log(user.password)
  if (passwordValid) {
    return response.send('Pass Valid')
  } else {
    return response.abort('Invalid credentials')
  }

  /*try {
    const token = await auth.use('api').generate(user)
    return response.json(token)
  } catch (err) {
    console.error(err)
    return response.internalServerError('Something went wrong')
  }*/`

**Comment by tomgobich** [comment]
What does your user model look like? Do you have the `AuthFinder` mixin being composed into it? If so, do note that mixin [will take care of password hashing](https://github.com/adonisjs/auth/blob/main/src/mixins/lucid.ts#L40-L50) via a `@beforeSave()` hook. It also provides a `verifyCredentials(uid, password)` method that accepts in the UID and password and will return back a user, if one was found, or throw an invalid credentials exception.

```ts
const user = await User.verifyCredentials(email, password)
```

https://docs.adonisjs.com/guides/verifying_user_credentials#using-the-auth-finder-mixin
